src/macaron/util.py

```diff
@@ -125,6 +125,80 @@ def send_get_http_raw(
     return response
 
 
+def send_post_http_raw(
```

Comment:
There is quite a bit of duplication between `send_post_http_raw`, `send_get_http_raw` and `send_get_http` (each has an implementation of error handling and retry logic), it would be good to refactor that at some point (but it doesn't have to be done as part of this change).